### PR TITLE
Add zod validation pipe

### PR DESCRIPTION
**Pull request by eliotfgn**

### CODE REVIEW

File: src/pipes/zod-validation.pipe.ts
Comment (YacineSteeve) [inline src/pipes/zod-validation.pipe.ts:1]: I think `validation.pipe.ts` is better for the file name.
As for the class to be named just `ValidationPipe`.

This way, we make the file names 'dependence-independant'. No need to change their name when changing the pipe implementation.

Comment (eliotfgn) [reply src/pipes/zod-validation.pipe.ts:1]: but who said we will change

Comment (YacineSteeve) [reply src/pipes/zod-validation.pipe.ts:1]: Prevention... Future... DESTINYYY

Comment (eliotfgn) [reply src/pipes/zod-validation.pipe.ts:1]: ok changed

**Review by YacineSteeve — APPROVED** [review]

**Pull request merged**